src/DateTime.cc

```diff
@@ -201,6 +201,42 @@ NsType taiToUtc(NsType nsecs) {
     return nsecs - leapNSecs;
 }
 
+/** Convert a time in nsec from any time scale to TAI
+ * \param[in] nsecs Number of nanoseconds since the epoch in TAI
+ * \param[in] scale The time scale (TAI, TT or UTC)
+ * \return Number of nanoseconds since the epoch in TAI
+ */
+long long nsecAnyToTai(long long nsecs, dafBase::DateTime::Timescale scale) {
+    if (scale == dafBase::DateTime::TAI) {
+        return nsecs;
+    } else if (scale == dafBase::DateTime::TT) {
+        return nsecs - TT_MINUS_TAI_NSECS;
+    } else if (scale == dafBase::DateTime::UTC) {
+        return utcToTai(nsecs);
+    }
```

Comment:
I think he means that as written the `if` statement can fall off the end, whereas if you use `switch` the compiler checks that every possible enum case has been included in the switch and warns if one is being ignored.